### ISSUE TITLE
Unable to open extension

### ISSUE THREAD

**Issue by oscarnylander** [tnesbitt210/interactive-smartlog-issues#9]
**User Report:**

Launching the extension leads to an error message: `Unable to fetch commits`.

Fwiw, I'm working in a *very* large repository.

---
**Debug Info:**

*commitsError*: 
```
RangeError: stdout maxBuffer length exceeded
    at new NodeError (node:internal/errors:405:5)
    at Socket.onChildStdout (node:child_process:491:14)
    at Socket.emit (node:events:517:28)
    at addChunk (node:internal/streams/readable:335:12)
    at readableAddChunk (node:internal/streams/readable:304:11)
    at Socket.Readable.push (node:internal/streams/readable:245:10)
    at Pipe.onStreamRead (node:internal/stream_base_commons:190:23)
```

*commitTreesLength*: 0

*uncommittedChangesLength*: 1

*outputChannelContents*: 
```
Setup analytics, session:  <anonymized>
Operating System: darwin 23.4.0
IGL Version: 0.1.8
Using cached repo info
[track] ClientConnection  
establish client connection for <anonymized>
Using cached repo info
Fetching commits...
Failed to fetch log: RangeError [ERR_CHILD_PROCESS_STDIO_MAXBUFFER]: stdout maxBuffer length exceeded
Loading commits failure
[track] InitialFetchSmartlogCommits InitialFetchSmartlogCommitsError 
[error] InitialFetchSmartlogCommits InitialFetchSmartlogCommitsError RangeError: stdout maxBuffer length exceeded
    at new NodeError (node:internal/errors:405:5)
    at Socket.onChildStdout (node:child_process:491:14)
    at Socket.emit (node:events:517:28)
    at addChunk (node:internal/streams/readable:335:12)
    at readableAddChunk (node:internal/streams/readable:304:11)
    at Socket.Readable.push (node:internal/streams/readable:245:10)
    at Pipe.onStreamRead (node:internal/stream_base_commons:190:23)
Sending commits to client
Fetching commits...
Failed to fetch log: RangeError [ERR_CHILD_PROCESS_STDIO_MAXBUFFER]: stdout maxBuffer length exceeded
Failed to fetch log: RangeError [ERR_CHILD_PROCESS_STDIO_MAXBUFFER]: stdout maxBuffer length exceeded
Failed to fetch log: RangeError [ERR_CHILD_PROCESS_STDIO_MAXBUFFER]: stdout maxBuffer length exceeded
Failed to fetch log: RangeError [ERR_CHILD_PROCESS_STDIO_MAXBUFFER]: stdout maxBuffer length exceeded
Failed to fetch log: RangeError [ERR_CHILD_PROCESS_STDIO_MAXBUFFER]: stdout maxBuffer length exceeded
Failed to fetch log: RangeError [ERR_CHILD_PROCESS_STDIO_MAXBUFFER]: stdout maxBuffer length exceeded
Failed to fetch log: RangeError [ERR_CHILD_PROCESS_STDIO_MAXBUFFER]: stdout maxBuffer length exceeded
Failed to fetch log: RangeError [ERR_CHILD_PROCESS_STDIO_MAXBUFFER]: stdout maxBuffer length exceeded
Failed to fetch log: RangeError [ERR_CHILD_PROCESS_STDIO_MAXBUFFER]: stdout maxBuffer length exceeded
Failed to fetch log: RangeError [ERR_CHILD_PROCESS_STDIO_MAXBUFFER]: stdout maxBuffer length exceeded
Failed to fetch log: RangeError [ERR_CHILD_PROCESS_STDIO_MAXBUFFER]: stdout maxBuffer length exceeded
Failed to fetch log: RangeError [ERR_CHILD_PROCESS_STDIO_MAXBUFFER]: stdout maxBuffer length exceeded
Failed to fetch log: RangeError [ERR_CHILD_PROCESS_STDIO_MAXBUFFER]: stdout maxBuffer length exceeded
Failed to fetch log: RangeError [ERR_CHILD_PROCESS_STDIO_MAXBUFFER]: stdout maxBuffer length exceeded
Failed to fetch log: RangeError [ERR_CHILD_PROCESS_STDIO_MAXBUFFER]: stdout maxBuffer length exceeded
Failed to fetch log: RangeError [ERR_CHILD_PROCESS_STDIO_MAXBUFFER]: stdout maxBuffer length exceeded
Failed to fetch log: RangeError [ERR_CHILD_PROCESS_STDIO_MAXBUFFER]: stdout maxBuffer length exceeded
Failed to fetch log: RangeError [ERR_CHILD_PROCESS_STDIO_MAXBUFFER]: stdout maxBuffer length exceeded
[track] TopLevelErrorShown CommitsFetchError 
[error] TopLevelErrorShown CommitsFetchError RangeError: stdout maxBuffer length exceeded
    at new NodeError (node:internal/errors:405:5)
    at Socket.onChildStdout (node:child_process:491:14)
    at Socket.emit (node:events:517:28)
    at addChunk (node:internal/streams/readable:335:12)
    at readableAddChunk (node:internal/streams/readable:304:11)
    at Socket.Readable.push (node:internal/streams/readable:245:10)
    at Pipe.onStreamRead (node:internal/stream_base_commons:190:23)
Failed to fetch log: RangeError [ERR_CHILD_PROCESS_STDIO_MAXBUFFER]: stdout maxBuffer length exceeded
Failed to fetch log: RangeError [ERR_CHILD_PROCESS_STDIO_MAXBUFFER]: stdout maxBuffer length exceeded
Failed to fetch log: RangeError [ERR_CHILD_PROCESS_STDIO_MAXBUFFER]: stdout maxBuffer length exceeded
Failed to fetch log: RangeError [ERR_CHILD_PROCESS_STDIO_MAXBUFFER]: stdout maxBuffer length exceeded
Failed to fetch log: RangeError [ERR_CHILD_PROCESS_STDIO_MAXBUFFER]: stdout maxBuffer length exceeded
Failed to fetch log: RangeError [ERR_CHILD_PROCESS_STDIO_MAXBUFFER]: stdout maxBuffer length exceeded
Failed to fetch log: RangeError [ERR_CHILD_PROCESS_STDIO_MAXBUFFER]: stdout maxBuffer length exceeded
Failed to fetch log: RangeError [ERR_CHILD_PROCESS_STDIO_MAXBUFFER]: stdout maxBuffer length exceeded
Failed to fetch log: RangeError [ERR_CHILD_PROCESS_STDIO_MAXBUFFER]: stdout maxBuffer length exceeded
Failed to fetch log: RangeError [ERR_CHILD_PROCESS_STDIO_MAXBUFFER]: stdout maxBuffer length exceeded
Failed to fetch log: RangeError [ERR_CHILD_PROCESS_STDIO_MAXBUFFER]: stdout maxBuffer length exceeded
Failed to fetch log: RangeError [ERR_CHILD_PROCESS_STDIO_MAXBUFFER]: stdout maxBuffer length exceeded
Failed to fetch log: RangeError [ERR_CHILD_PROCESS_STDIO_MAXBUFFER]: stdout maxBuffer length exceeded
Failed to fetch log: RangeError [ERR_CHILD_PROCESS_STDIO_MAXBUFFER]: stdout maxBuffer length exceeded
Failed to fetch log: RangeError [ERR_CHILD_PROCESS_STDIO_MAXBUFFER]: stdout maxBuffer length exceeded
Failed to fetch log: RangeError [ERR_CHILD_PROCESS_STDIO_MAXBUFFER]: stdout maxBuffer length exceeded
Failed to fetch log: RangeError [ERR_CHILD_PROCESS_STDIO_MAXBUFFER]: stdout maxBuffer length exceeded
Failed to fetch log: RangeError [ERR_CHILD_PROCESS_STDIO_MAXBUFFER]: stdout maxBuffer length exceeded
Failed to fetch log: RangeError [ERR_CHILD_PROCESS_STDIO_MAXBUFFER]: stdout maxBuffer length exceeded
Failed to fetch log: RangeError [ERR_CHILD_PROCESS_STDIO_MAXBUFFER]: stdout maxBuffer length exceeded
Failed to fetch log: RangeError [ERR_CHILD_PROCESS_STDIO_MAXBUFFER]: stdout maxBuffer length exceeded
Failed to fetch log: RangeError [ERR_CHILD_PROCESS_STDIO_MAXBUFFER]: stdout maxBuffer length exceeded
Failed to fetch log: RangeError [ERR_CHILD_PROCESS_STDIO_MAXBUFFER]: stdout maxBuffer length exceeded
Failed to fetch log: RangeError [ERR_CHILD_PROCESS_STDIO_MAXBUFFER]: stdout maxBuffer length exceeded
Failed to fetch log: RangeError [ERR_CHILD_PROCESS_STDIO_MAXBUFFER]: stdout maxBuffer length exceeded
Failed to fetch log: RangeError [ERR_CHILD_PROCESS_STDIO_MAXBUFFER]: stdout maxBuffer length exceeded
Failed to fetch log: RangeError [ERR_CHILD_PROCESS_STDIO_MAXBUFFER]: stdout maxBuffer length exceeded
Failed to fetch log: RangeError [ERR_CHILD_PROCESS_STDIO_MAXBUFFER]: stdout maxBuffer length exceeded
Failed to fetch log: RangeError [ERR_CHILD_PROCESS_STDIO_MAXBUFFER]: stdout maxBuffer length exceeded
[track] ClientDisconnection  
Failed to fetch log: RangeError [ERR_CHILD_PROCESS_STDIO_MAXBUFFER]: stdout maxBuffer length exceeded
[track] ClientConnection  
establish client connection for <anonymized>
Using cached repo info
Fetching commits...
Failed to fetch log: RangeError [ERR_CHILD_PROCESS_STDIO_MAXBUFFER]: stdout maxBuffer length exceeded
Failed to fetch log: RangeError [ERR_CHILD_PROCESS_STDIO_MAXBUFFER]: stdout maxBuffer length exceeded
Failed to fetch log: RangeError [ERR_CHILD_PROCESS_STDIO_MAXBUFFER]: stdout maxBuffer length exceeded
Failed to fetch log: RangeError [ERR_CHILD_PROCESS_STDIO_MAXBUFFER]: stdout maxBuffer length exceeded
Failed to fetch log: RangeError [ERR_CHILD_PROCESS_STDIO_MAXBUFFER]: stdout maxBuffer length exceeded
Failed to fetch log: RangeError [ERR_CHILD_PROCESS_STDIO_MAXBUFFER]: stdout maxBuffer length exceeded
Failed to fetch log: RangeError [ERR_CHILD_PROCESS_STDIO_MAXBUFFER]: stdout maxBuffer length exceeded
Failed to fetch log: RangeError [ERR_CHILD_PROCESS_STDIO_MAXBUFFER]: stdout maxBuffer length exceeded
Failed to fetch log: RangeError [ERR_CHILD_PROCESS_STDIO_MAXBUFFER]: stdout maxBuffer length exceeded
Failed to fetch log: RangeError [ERR_CHILD_PROCESS_STDIO_MAXBUFFER]: stdout maxBuffer length exceeded
Failed to fetch log: RangeError [ERR_CHILD_PROCESS_STDIO_MAXBUFFER]: stdout maxBuffer length exceeded
Failed to fetch log: RangeError [ERR_CHILD_PROCESS_STDIO_MAXBUFFER]: stdout maxBuffer length exceeded
Failed to fetch log: RangeError [ERR_CHILD_PROCESS_STDIO_MAXBUFFER]: stdout maxBuffer length exceeded
Failed to fetch log: RangeError [ERR_CHILD_PROCESS_STDIO_MAXBUFFER]: stdout maxBuffer length exceeded
Failed to fetch log: RangeError [ERR_CHILD_PROCESS_STDIO_MAXBUFFER]: stdout maxBuffer length exceeded
Failed to fetch log: RangeError [ERR_CHILD_PROCESS_STDIO_MAXBUFFER]: stdout maxBuffer length exceeded
Failed to fetch log: RangeError [ERR_CHILD_PROCESS_STDIO_MAXBUFFER]: stdout maxBuffer length exceeded
Failed to fetch log: RangeError [ERR_CHILD_PROCESS_STDIO_MAXBUFFER]: stdout maxBuffer length exceeded
Failed to fetch log: RangeError [ERR_CHILD_PROCESS_STDIO_MAXBUFFER]: stdout maxBuffer length exceeded
Failed to fetch log: RangeError [ERR_CHILD_PROCESS_STDIO_MAXBUFFER]: stdout maxBuffer length exceeded
Failed to fetch log: RangeError [ERR_CHILD_PROCESS_STDIO_MAXBUFFER]: stdout maxBuffer length exceeded
Failed to fetch log: RangeError [ERR_CHILD_PROCESS_STDIO_MAXBUFFER]: stdout maxBuffer length exceeded
Failed to fetch log: RangeError [ERR_CHILD_PROCESS_STDIO_MAXBUFFER]: stdout maxBuffer length exceeded
Failed to fetch log: RangeError [ERR_CHILD_PROCESS_STDIO_MAXBUFFER]: stdout maxBuffer length exceeded
Failed to fetch log: RangeError [ERR_CHILD_PROCESS_STDIO_MAXBUFFER]: stdout maxBuffer length exceeded
Failed to fetch log: RangeError [ERR_CHILD_PROCESS_STDIO_MAXBUFFER]: stdout maxBuffer length exceeded
Failed to fetch log: RangeError [ERR_CHILD_PROCESS_STDIO_MAXBUFFER]: stdout maxBuffer length exceeded
Failed to fetch log: RangeError [ERR_CHILD_PROCESS_STDIO_MAXBUFFER]: stdout maxBuffer length exceeded
Failed to fetch log: RangeError [ERR_CHILD_PROCESS_STDIO_MAXBUFFER]: stdout maxBuffer length exceeded
Failed to fetch log: RangeError [ERR_CHILD_PROCESS_STDIO_MAXBUFFER]: stdout maxBuffer length exceeded
Failed to fetch log: RangeError [ERR_CHILD_PROCESS_STDIO_MAXBUFFER]: stdout maxBuffer length exceeded
Failed to fetch log: RangeError [ERR_CHILD_PROCESS_STDIO_MAXBUFFER]: stdout maxBuffer length exceeded
Failed to fetch log: RangeError [ERR_CHILD_PROCESS_STDIO_MAXBUFFER]: stdout maxBuffer length exceeded
Failed to fetch log: RangeError [ERR_CHILD_PROCESS_STDIO_MAXBUFFER]: stdout maxBuffer length exceeded
Failed to fetch log: RangeError [ERR_CHILD_PROCESS_STDIO_MAXBUFFER]: stdout maxBuffer length exceeded
Failed to fetch log: RangeError [ERR_CHILD_PROCESS_STDIO_MAXBUFFER]: stdout maxBuffer length exceeded
Failed to fetch log: RangeError [ERR_CHILD_PROCESS_STDIO_MAXBUFFER]: stdout maxBuffer length exceeded
Failed to fetch log: RangeError [ERR_CHILD_PROCESS_STDIO_MAXBUFFER]: stdout maxBuffer length exceeded
Failed to fetch log: RangeError [ERR_CHILD_PROCESS_STDIO_MAXBUFFER]: stdout maxBuffer length exceeded
Failed to fetch log: RangeError [ERR_CHILD_PROCESS_STDIO_MAXBUFFER]: stdout maxBuffer length exceeded
Failed to fetch log: RangeError [ERR_CHILD_PROCESS_STDIO_MAXBUFFER]: stdout maxBuffer length exceeded
Failed to fetch log: RangeError [ERR_CHILD_PROCESS_STDIO_MAXBUFFER]: stdout maxBuffer length exceeded
Failed to fetch log: RangeError [ERR_CHILD_PROCESS_STDIO_MAXBUFFER]: stdout maxBuffer length exceeded
Failed to fetch log: RangeError [ERR_CHILD_PROCESS_STDIO_MAXBUFFER]: stdout maxBuffer length exceeded
Failed to fetch log: RangeError [ERR_CHILD_PROCESS_STDIO_MAXBUFFER]: stdout maxBuffer length exceeded
Failed to fetch log: RangeError [ERR_CHILD_PROCESS_STDIO_MAXBUFFER]: stdout maxBuffer length exceeded
Failed to fetch log: RangeError [ERR_CHILD_PROCESS_STDIO_MAXBUFFER]: stdout maxBuffer length exceeded
Loading commits failure
[track] InitialFetchSmartlogCommits InitialFetchSmartlogCommitsError 
[error] InitialFetchSmartlogCommits InitialFetchSmartlogCommitsError RangeError: stdout maxBuffer length exceeded
    at new NodeError (node:internal/errors:405:5)
    at Socket.onChildStdout (node:child_process:491:14)
    at Socket.emit (node:events:517:28)
    at addChunk (node:internal/streams/readable:335:12)
    at readableAddChunk (node:internal/streams/readable:304:11)
    at Socket.Readable.push (node:internal/streams/readable:245:10)
    at Pipe.onStreamRead (node:internal/stream_base_commons:190:23)
Sending commits to client
Fetching commits...
Failed to fetch log: RangeError [ERR_CHILD_PROCESS_STDIO_MAXBUFFER]: stdout maxBuffer length exceeded
[track] TopLevelErrorShown CommitsFetchError 
[error] TopLevelErrorShown CommitsFetchError RangeError: stdout maxBuffer length exceeded
    at new NodeError (node:internal/errors:405:5)
    at Socket.onChildStdout (node:child_process:491:14)
    at Socket.emit (node:events:517:28)
    at addChunk (node:internal/streams/readable:335:12)
    at readableAddChunk (node:internal/streams/readable:304:11)
    at Socket.Readable.push (node:internal/streams/readable:245:10)
    at Pipe.onStreamRead (node:internal/stream_base_commons:190:23)
Failed to fetch log: RangeError [ERR_CHILD_PROCESS_STDIO_MAXBUFFER]: stdout maxBuffer length exceeded
Failed to fetch log: RangeError [ERR_CHILD_PROCESS_STDIO_MAXBUFFER]: stdout maxBuffer length exceeded
<truncated>
```

**Comment by oscarnylander** [comment]
>The error you're encountering seems to suggest that your git log output is too large for stdout, which would suggest that you have a ton of commits between the branch head and its intersection point with origin/master (or what ever the name of the primary remote branch is)

Could it be that I have stale branches locally? I'm not particularly good at cleaning up my local branches, which leads to `git branch | wc -l` returning `123`.

Let me see if running some cleanup fixes the issue.

**Comment by oscarnylander** [comment]
Fwiw, my `master` was outdated by 4000+ commits, probably because I've stopped using it as my basis to branch off in favour of another branch.

**Comment by tnesbitt210** [comment]
Ah ok, that's probably the issue. We're trying to pipe 4000+ commits (for each branch!) into stdout

I can see if there's a way for me to bump the stdout limit, or maybe implement some sort of pagination over the logs. But this might be one of the more challenging things to fix.

**Comment by oscarnylander** [comment]
Pulling down `master` didn't resolve the issue though, perhaps because there's a lot of commits between the newly updated `master` and stale branches?

**Comment by tnesbitt210** [comment]
Yeah exactly. Those branches still have 4000+ commits between their head & where they intersect with `origin/master`.

I didn't really build with this scenario in mind, but it would be nice to fix it nonetheless. Unfortunately, there's probably not a way for you to use the extension with this particular repo until I fix it :/

**Comment by oscarnylander** [comment]
I'll have to try to clean up those stale branches I think. It's a bit more tricky on account of the branches being squash-merged, hence I can't really figure out which are merged in a simple way either. Darn.

**Comment by oscarnylander** [comment]
Great extension though btw, really liking it so far!

**Comment by tnesbitt210** [comment]
I'll look later today and see if I can get a quick & dirty fix for you :) No promises, but we'll see.

> Great extension though btw, really liking it so far!

And thanks!

**Comment by oscarnylander** [comment]
I did some serious spring cleaning of my local branches, but unfortunately I still have issues despite having removed anything even remotely old-looking in my `git branch`-output.

Do you have any helpful commands I could call to determine any 'offending branches'?

**Comment by tnesbitt210** [comment]
`git branch --format '%(refname:short)' | xargs -I {} sh -c 'echo -n "{},"; git rev-list --count origin/master..{}' `

^ assumes that `origin/master` is the name of the primary remote branch

It should spit out each branch, plus how many commits are on that branch between origin/master and the branch head.

**Comment by tnesbitt210** [comment]
Hey @oscarnylander , I just published v0.1.9 - which should hopefully fix your issue! 🙂

This version increases the default stdout buffer size from 1MB to 5MB, and it also allows you to configure the max buffer size via the VSCode setting `interactiveGitLog.maxBufferMB` (values less than 1 will be rounded up to 1)

<img width="331" alt="Screenshot 2024-04-24 at 8 58 20 PM" src="https://github.com/tnesbitt210/interactive-smartlog-issues/assets/10647853/48d1eede-cf4c-4522-8ab3-dbd9b9300c30">

I'm pretty curious if
1.  the fix works
2. the UI is even feels usable to you when it displays 4000+ commits
3. the load times tank for this repo because it needs to display so many commits

So, definitely follow up and let me know!

Also please feel free to reach out any time if you have any other bugs or feature requests! Very fun to hear from someone who is using & enjoying the extension. 🎉

**Comment by oscarnylander** [comment]
I managed to solve the issue - here's my findings:

- Updating the extension without changing any settings did not resolve the issue
- Increasing the buffer size to a large value (50 MB) led to an endless spinner
- Using the provided command against the remote branch that I'm primarily working against showed a reasonable 'commit-distance' (>30) for all local branches
- Using the command against my previously used but now unused remote showed a 'commit-distance' of >30k

For context: I use a two-remote setup for development:
- One for my own fork of the repository - this is where I push my branches to
- One for the mainline repository - I use this exclusively for branching off of

I assume that the default settings primarily look at `origin` for getting the intermediate commits? Anyway, I solved the issue by updating the origin's master branch to be more up-to-date with the mainline repository. I also noticed that there's a remote selector available, and I've switched that one to target the mainline repository instead of my fork, I suppose that might avoid the issue in the future as well.
